tests/test_class_new.rs

```diff
@@ -204,3 +204,29 @@ fn new_with_custom_error() {
         assert_eq!(err.to_string(), "ValueError: custom error");
     });
 }
+
+#[pyclass]
+#[derive(Clone, Debug)]
+struct NewWithClassMethod;
+
+#[pymethods]
+impl NewWithClassMethod {
+    #[new]
+    #[classmethod]
+    fn new(cls: &PyType) -> PyResult<Self> {
+        assert!(cls.is_subclass_of::<NewWithClassMethod>()?);
+        Ok(Self)
+    }
+}
```

Comment:
It would be nice to have a test which extends this struct to show the subtyping works as expected. Maybe the method could take an argument of the type to expect:

```rust
    #[new]
     #[classmethod]
     fn new(cls: &PyType, expected_type: &PyType) -> PyResult<Self> {
         assert!(cls.is(expected_type));
         Ok(Self)
     }
```

It might be easier to do the subtyping in Python by putting this test in the `pytests/` test suite.